### PR TITLE
Autocomplete usage and typedef clause

### PR DESCRIPTION
**Pull request by GabrielPdaC**

### CODE REVIEW

**Review by Lucasspaniol — CHANGES_REQUESTED** [review]
Very good, just a few adjustments to anticipate use in other COBOL development standards

File: src/lsp/completion/CobolVariable.ts [hunk 1]
```diff
@@ -277,16 +280,28 @@ export class CobolVariable {
...
+        return decRegExp.test(picture);
+    }
+
+    /**
+     * Returns true if the picture represents a typedef variable
+    *
+    * @param picture variable picture
+    */
+   private static isTypedef(picture: string): boolean {
+        const decRegExp: RegExp = /^[A-Za-z][A-Za-z0-9_]*-[A-Za-z0-9_-]*$/i;
```
[the remark this answers is not in the document]
Comment (GabrielPdaC) [reply src/lsp/completion/CobolVariable.ts]: Adjusted according to your suggestion

File: src/lsp/completion/CobolVariable.ts [hunk 2]
```diff
@@ -277,16 +280,28 @@ export class CobolVariable {
      * @param picture variable picture
      */
     private static isAlphanumeric(picture: string): boolean {
-        return picture.includes("X") || picture.includes("/");
+        const alfaRegExp: RegExp = /x\(\d+\)/i;
```
[the remark this answers is not in the document]
Comment (GabrielPdaC) [reply src/lsp/completion/CobolVariable.ts]: Adjusted pic x regex

**Review by Lucasspaniol — APPROVED** [review]
👍

**Pull request merged**